include/ua_plugin_securitypolicy.h

```diff
@@ -0,0 +1,424 @@
+                                        size_t bytesToWrite,
+                                        UA_Byte *paddingSize,
+                                        UA_Byte *extraPaddingSize);
+
+    const UA_SecurityPolicySigningModule signingModule;
+
+    const size_t signingKeyLength;
+    const size_t encryptingKeyLength;
+    const size_t encryptingBlockSize;
+} UA_SecurityPolicySymmetricModule;
+
+struct UA_Endpoint_SecurityContext {
+    UA_StatusCode (*const init)(const UA_SecurityPolicy *securityPolicy,
+                                const void *initData,
+                                void **pp_contextData);
+
+    UA_StatusCode (*const deleteMembers)(const UA_SecurityPolicy *securityPolicy,
+                                         void *endpointContext);
+
+    UA_StatusCode (*const setLocalPrivateKey)(const UA_SecurityPolicy *securityPolicy,
+                                              const UA_ByteString *privateKey,
+                                              void *endpointContext);
+
+    UA_StatusCode (*const setServerCertificate)(const UA_SecurityPolicy *securityPolicy,
+                                                const UA_ByteString *certificate,
+                                                void *endpointContext);
+
+    const UA_ByteString *(*const getServerCertificate)(const UA_SecurityPolicy *securityPolicy,
+                                                       const void *endpointContext);
+
+    UA_StatusCode (*const setCertificateTrustList)(const UA_SecurityPolicy *securityPolicy,
+                                                   const UA_ByteString *trustList,
+                                                   void *endpointContext);
+
+    UA_StatusCode (*const setCertificateRevocationList)(const UA_SecurityPolicy *securityPolicy,
+                                                        const UA_ByteString *revocationList,
+                                                        void *endpointContext);
+
+    /**
+    * Gets the signature size that depends on the local private key.
+    * This will return 0 as long as no remote certificate was set.
+    */
+    size_t (*const getLocalAsymSignatureSize)(const UA_SecurityPolicy *securityPolicy,
+                                              const void *endpointContext);
+
+    /**
+     * \brief Compares the supplied certificate with the certificate in the endpoit context.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param endpointContext the endpoint context data that contains the certificate to compare to.
+     * \param certificateThumbprint the certificate thumbprint to compare to the one stored in the context.
+     * \return if the thumbprints match UA_STATUSCODE_GOOD is returned. If they don't match
+     *         or an errror occured an error code is returned.
+     */
+    UA_StatusCode (*const compareCertificateThumbprint)(const UA_SecurityPolicy *securityPolicy,

```

Comment:
Same reasoning as above, with the difference, that this only makes sense for the endpoint context/asymmetric module. imho this should stay part of the endpoint context.